src/IO/S3/Requests.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AWS_S3
+
+#include <IO/S3/URI.h>
+
+#include <aws/core/endpoint/EndpointParameter.h>
+#include <aws/s3/model/HeadObjectRequest.h>
+#include <aws/s3/model/ListObjectsV2Request.h>
+#include <aws/s3/model/ListObjectsRequest.h>
+#include <aws/s3/model/GetObjectRequest.h>
+#include <aws/s3/model/AbortMultipartUploadRequest.h>
+#include <aws/s3/model/CreateMultipartUploadRequest.h>
+#include <aws/s3/model/CompleteMultipartUploadRequest.h>
+#include <aws/s3/model/CopyObjectRequest.h>
+#include <aws/s3/model/PutObjectRequest.h>
+#include <aws/s3/model/UploadPartRequest.h>
+#include <aws/s3/model/UploadPartCopyRequest.h>
+#include <aws/s3/model/DeleteObjectRequest.h>
+#include <aws/s3/model/DeleteObjectsRequest.h>
+
+namespace DB::S3
+{
+
+namespace Model = Aws::S3::Model;
+
+template <typename BaseRequest>
+class ExtendedRequest : public BaseRequest
+{
+public:
+    Aws::Endpoint::EndpointParameters GetEndpointContextParams() const override
+    {
+        auto params = BaseRequest::GetEndpointContextParams();
+        if (!region_override.empty())
+            params.emplace_back("Region", region_override);
```

Comment:
Is it possible that `BaseRequest::GetEndpointContextParams()` already has this override?

Reply:
Could be possible but the params are processed sequentially where the latest will always override the previous.
Also, the params are fetched from 3 different sources where `BaseRequest::GetEndpointContextParams` is the last one so we are sure it will always win. An assumption we need to be careful of if they change it in SDK (don't see a reason to do it but it could happen). I need to add a comment explaining exactly this.